### DISCUSSION TITLE
Connecting SOC with external premission adminsitrated elasticsearch

### DISCUSSION THREAD

**Discussion by thealonzo** [Security-Onion-Solutions/securityonion#8190] [Unsupported Versions] [answered]
Hey,

I'm using SOC v 2.3.60 and I want to connect it with an external elasticsearch with different user premissions, administrated by the external elasticsearch.

I succedded running it with a non administrated elasticsearch.

I read a bit of the code of SOC and I found in the project securityonion-soc in the path server/modules/elastic/elastictransport.go the following code:

    **_if user.SearchUsername != "" {
      username = user.SearchUsername
    }
    req.Header.Set("es-security-runas-user", username)_**

I think it might have somthing to do with the SearchUsername defined for the user created.
I just don't know how and where can I configure it. I looked for it

If I can configure the SearchUsername to work with my external managed elasticserach it will help me resolve the issue.

Thanks a lot,
Alon

**Comment by weslambert** [comment, marked as answer]
While it's technically possible, we don't provide support or guarantee proper operation for/when connecting to an external Elasticsearch instance, especially one that is governed by its own set of permissions.
